src/zarr/testing/store.py

```diff
@@ -192,6 +192,13 @@ async def test_list_dir(self, store: S) -> None:
         assert [k async for k in store.list_dir("foo")] == []
         await store.set("foo/zarr.json", Buffer.from_bytes(b"bar"))
         await store.set("foo/c/1", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/1", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/2", Buffer.from_bytes(b"\x01"))
+        await store.set("foo/c/d/3", Buffer.from_bytes(b"\x01"))
```

Comment:
I don't think its super material to the test but I believe the intent of this test was to have `foo` be an array `c` be the subdirectory containing chunks. Perhaps it would be better to change this to something more like: 
```suggestion
        await store.set("foo/bar/c/1", Buffer.from_bytes(b"\x01"))
        await store.set("foo/bar/c/2", Buffer.from_bytes(b"\x01"))
        await store.set("foo/spam/c/1", Buffer.from_bytes(b"\x01"))
        await store.set("foo/spam/c/2", Buffer.from_bytes(b"\x01"))
```

Reply:
Right... IIUC, if you have a key at `foo/zarr.json` with a `node_type="array"`, that implies you *won't* have any keys at `foo/bar/zarr.json` (a group or a node), since arrays can't have children?

I tried to update this in 7414b1064311fd17f41a40c97fcf96b2299ac29c, to make sure that the test store we construct is at least compatible with how a Zarr hierarchy would be laid out.

Reply:
Looks great!